### PR TITLE
Implement node attributes

### PR DESCRIPTION
**Pull request by Pyrofab**
Third party attributes, to unlock the *true potential* (at least the first one) of fiber's schemas.

Implementation: ~~it's just another `ConfigNode`, code reusability good.~~ terrible idea, it is now a `Map<Identifier, ConfigAttribute>`. The `ConfigAttribute` interface is currently just an empty subtype of Property, although we may decide to add relevant methods there.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java [hunk 1]
```diff
@@ -50,6 +56,18 @@ public ConfigBranch getParent() {
...
+    public Map<Identifier, ConfigAttribute<?>> getAttributes() {
+        return this.attributes;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <A> ConfigAttribute<A> getOrCreateAttribute(Identifier id, @Nonnull A defaultValue) {
+        return (ConfigAttribute<A>) getAttributes()
+                .computeIfAbsent(id, i -> new ConfigAttributeImpl<>(defaultValue, (Class<A>) defaultValue.getClass()));
+    }
```
Comment (kvverti) [inline src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java:69]: Careful, if the default value is of a subtype this won't work (hence the unchecked warning).

File: src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java [hunk 2]
```diff
@@ -17,6 +20,22 @@
...
+     *
+     * <p> As the returned data structure is shared by every attribute source,
+     * attributes should be grouped by namespace.
+     *
+     * @return this node's configurable attributes
+     */
+    Map<Identifier, ConfigAttribute<?>> getAttributes();
+
+    <A> ConfigAttribute<A> getOrCreateAttribute(Identifier id, @Nonnull A defaultValue);
+
```
Comment (kvverti) [inline src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java:38]: See comment below about subtyping. There is no reliable way to get the class object for a given type from an object compatible with that type.

File: src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java (hunk 1 above)
Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java:69]: oh no, how could I forget about that

File: src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java [hunk 3]
```diff
@@ -17,6 +20,44 @@
...
+     * <p> Attributes store metadata pertaining to the node itself, rather than its value.
+     * The returned map can be mutated by third parties to supplement the default node metadata.
+     * Examples of attributes include translation keys or rendering information.
+     *
+     * <p> As the returned data structure is shared by every attribute source,
+     * attributes should be grouped by namespace.
+     *
+     * @return this node's configurable attributes
+     */
+    Map<Identifier, ConfigAttribute<?>> getAttributes();
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java:35]: `Identifier`s aren't strings and shouldn't be treated as such! Even if you create two identical `Identifier`s, you can store both in this map and have them point to different things.

This'll produce some unintended side-effects:
```java
attributes.put(new Identifier("fiber", "identifier"), someAttribute);

// Later, we want to get the 'fiber:identifier' attribute...
attributes.get(new Identifier("fiber", "identifier")) == null; // true 
```

Instead, consider
* Having `String` keys instead
* Changing the `getAttributeValue` logic to look for a matching identifier instead, and not the exact same identifier

Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java:35]: What prevents us from providing `Identifier#equals` ?

Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/fiber/tree/ConfigNode.java:35]: Absolutely nothing and I'm a dork

File: src/main/java/me/zeroeightsix/fiber/tree/ConfigAttributeImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    public T getValue() {
+        return this.value;
+    }
+
+    @Override
+    public Class<T> getType() {
+        return this.type;
+    }
```
[the remark this answers is not in the document]
Comment (Pyrofab) [reply src/main/java/me/zeroeightsix/fiber/tree/ConfigAttributeImpl.java:26]: This method is exposed through the `HasValue` interface, which `Property` extends.

**Pull request merged**